buildSrc/src/main/java/org/elasticsearch/gradle/AbstractLazyPropertyCollection.java

```diff
@@ -0,0 +1,27 @@
+package org.elasticsearch.gradle;
+
+import java.util.List;
+
+public abstract class AbstractLazyPropertyCollection {
```

Comment:
Do you mean for `getNormalizedCollection()`? If so we cannot because implementations might map the value type to _any_ other type for the purposes of input snapshotting. Since this is only used by Gradle for input snapshotting, we only really care about the type at runtime.